lib/mem.h

```diff
@@ -63,7 +63,17 @@ extern "C" {
 /*-**************************************************************
 *  Basic Types
 *****************************************************************/
-#if defined (__cplusplus) || (defined (__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L) /* C99 */)
+#if defined(_MSC_VER) && (_MSC_VER < 1400)     /* For VC6 and other legacy versions */
+  typedef unsigned char       BYTE;
+  typedef unsigned short      U16;
+  typedef   signed short      S16;
+  typedef unsigned int        U32;
+  typedef   signed int        S32;
+  typedef unsigned __int64    U64;
+  typedef   signed __int64    S64;
+  #define LIT_S64(x) x##i64
```

Comment:
In LZ compression, literals designate bytes which won't be regenerated using an LZ match.
Since it's not the point here, you might refer to another concept I'm not aware of.
Anyway, in the context of LZ compression, it will be confusing.

The objective of these macros is to write 64-bits explicit constants, that are currently written `12345ULL`, in a format which VC6 is able to understand.
So I would suggest as names `CONST_U64()` and `CONST_S64()`.